great_expectations/datasource/fluent/config_str.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from pydantic import SecretStr
+
+from great_expectations.core.config_substitutor import TEMPLATE_STR_REGEX
+
+if TYPE_CHECKING:
+    from great_expectations.core.config_provider import _ConfigurationProvider
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ConfigStr(SecretStr):
+    def __init__(
+        self,
+        template_str: str,
+    ) -> None:
+        self.template_str: str = template_str
+
+        # self.config_provider - allows the config provider to manually attached to the field.
```

Comment:
Yes, I would pick one paradigm or the other. Either pass it in, in the constructor and use it throughout the class or make the user pass it in each time when calling `get_config_value` and don't have this member variable at all.